drivers/dma/CMakeLists.txt

```diff
@@ -7,6 +7,7 @@ zephyr_library_sources_ifdef(CONFIG_DMA_STM32U5	dma_stm32u5.c)
 zephyr_library_sources_ifdef(CONFIG_DMA_STM32		dma_stm32.c)
 zephyr_library_sources_ifdef(CONFIG_DMA_STM32_V1	dma_stm32_v1.c)
 zephyr_library_sources_ifdef(CONFIG_DMA_STM32_V2	dma_stm32_v2.c)
+zephyr_library_sources_ifdef(CONFIG_BDMA_STM32		bdma_stm32.c)
```

Comment:
DMA_STM32_BDMA as a Kconfig symbol and dma_stm32_bdma.c as a file name would better match the existing convention